lib/internal/handlers.js

```diff
@@ -0,0 +1,31 @@
+"use strict";
+
+// Adapted from wpt tools
+// https://github.com/web-platform-tests/wpt/blob/master/tools/wptserve/wptserve/handlers.py
+
+const path = require("path");
+const { parse: parseUrl } = require("url");
```

Comment:
Can you use the modern WHATWG URL parser, so `const { URL } = require("url")`? That may also do decoding for you, not sure.

Reply:
Sure, I can do that.

[This would require Node 7 or higher](https://nodejs.org/api/url.html#url_class_url), perhaps we should signal the minimum Node version somewhere, e.g. in [`engines.node` of `package.json`](https://docs.npmjs.com/files/package.json#engines)?

I noticed that there's a `.travis.yml` file in the project that targets Node 4, but there aren't any tests to run on Travis. 🤷‍♂️

Reply:
Haha, oh dear. Yeah, let's bump to Node 7 min version, and I'll release this as a new major.

Reply:
Hmm, the WHATWG URL parser can't handle relative URLs without passing an absolute base URL. I'll need to check what base URL to use for every parse. I'll look into it.